### PR TITLE
KURSP-627: New deploy pipeline

### PR DESCRIPTION
**Pull request by audunjl**
* Upgrade PHP version from 7 to 8
* Upgrade Laravel version from 8 to 9
* Upgrade Vue version from 2 to 3
* Change from using Laravel Mix to Vite
* Dockerfile for local development
* Dockerfile for production in Azure App Service (Linux container)
* Configuration for CI/CD deploy pipeline
* Cronjob for running Laravel Scheduler

### CODE REVIEW

File: app/Http/Controllers/CommandController.php [hunk 1]
```diff
@@ -22,4 +22,10 @@ public function run_scheduler()
         Artisan::call('schedule:run');
         return 'OK';
     }
+
+    public function run_fetch_nsr()
+    {
+        Artisan::call('fetch_from:nsr');
```
Comment (audunjl) [inline app/Http/Controllers/CommandController.php:28]: @audunjl Denne trengs ikke lenger

File: prod.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+# SERVER
+FROM php:8.1-fpm
+
+#Install packages
+RUN apt-get update && apt-get install -y \
+    cron \
+    vim \
+	libpng-dev \
```
Comment (audunjl) [inline prod.Dockerfile:26]: @audunjl fiks formatering

File: routes/api.php [hunk 3]
```diff
@@ -14,6 +14,8 @@
 
 Route::post('/run_scheduler', 'CommandController@run_scheduler')->middleware('token_auth');
 
+Route::post('/run_fetch_nsr', 'CommandController@run_fetch_nsr')->middleware('token_auth');
+
```
Comment (audunjl) [inline routes/api.php:18]: @audunjl trenger ikke denne

File: resources/js/components/GroupSelector.vue [hunk 4]
```diff
@@ -179,6 +154,7 @@
         delete this.selectedgroups.community;
         delete this.selectedgroups.institution;
         this.selectedgroups.county = this.getCountyGroup();
+        console.log(this.chosenCounty);
```
Comment (audunjl) [inline resources/js/components/GroupSelector.vue:157]: @theahthodesen OK å fjerne denne linja?

Comment (madsenandreas) [inline resources/js/components/GroupSelector.vue:157]: Debug melding?

File: resources/js/components/RoleSelector.vue [hunk 5]
```diff
@@ -38,14 +39,13 @@
     created() {
       this.wantToBePrincipal = this.isPrincipal;
     },
+    emits: ['update:value'],
     watch: {
       isPrincipal(value) {
         this.wantToBePrincipal = value;
+        console.log(this.isPrincipal)
```
Comment (madsenandreas) [inline resources/js/components/RoleSelector.vue:46]: Debug melding?

File: resources/js/views/GroupEnrollView.vue [hunk 6]
```diff
@@ -237,6 +215,7 @@
           this.iframeresize();
         }
         this.currentGroupsLoaded = true;
+        console.log(this.currentGroups)
```
Comment (madsenandreas) [inline resources/js/views/GroupEnrollView.vue:218]: debug?

File: routes/api.php (hunk 3 above)
Comment (audunjl) [reply routes/api.php]: Fikset

File: prod.Dockerfile (hunk 2 above)
Comment (audunjl) [reply prod.Dockerfile]: Fikset

File: resources/js/components/RoleSelector.vue (hunk 5 above)
Comment (audunjl) [reply resources/js/components/RoleSelector.vue:46]: Hørte med Thea, OK å fjerne

File: resources/js/components/GroupSelector.vue (hunk 4 above)
Comment (audunjl) [reply resources/js/components/GroupSelector.vue:157]: Hørte med Thea, OK å fjerne

File: resources/js/views/GroupEnrollView.vue (hunk 6 above)
Comment (audunjl) [reply resources/js/views/GroupEnrollView.vue:218]: Hørte med Thea, OK å fjerne

File: resources/js/components/GroupSelector.vue (hunk 4 above)
Comment (audunjl) [reply resources/js/components/GroupSelector.vue]: Fikset

Comment (audunjl) [reply resources/js/components/GroupSelector.vue]: Fikset

File: resources/js/views/GroupEnrollView.vue (hunk 6 above)
Comment (audunjl) [reply resources/js/views/GroupEnrollView.vue]: Fikset

File: resources/js/components/RoleSelector.vue (hunk 5 above)
Comment (audunjl) [reply resources/js/components/RoleSelector.vue]: Fikset

File: app/Http/Controllers/CommandController.php (hunk 1 above)
Comment (audunjl) [reply app/Http/Controllers/CommandController.php]: Fikset

**Review by manilpit — APPROVED** [review]

**Pull request merged**